httputil/json.go

```diff
@@ -25,3 +25,12 @@ func WriteJSONResponse(w http.ResponseWriter, code int, resp interface{}) error
 	}
 	return nil
 }
+
+func WriteError(w http.ResponseWriter, code int, msg string) error {
```

Comment:
I'm not entirely convinced this should go in as-is. Different APIs may require different response formats.